comp/syntheticstestscheduler/common/data.go

```diff
@@ -0,0 +1,161 @@
+type SubType string
+
+const (
+	// SubTypeUDP represents a udp network test.
+	SubTypeUDP SubType = "udp"
+	// SubTypeTCP represents a tcp network test.
+	SubTypeTCP SubType = "tcp"
+	// SubTypeICMP represents a icmp network test.
+	SubTypeICMP SubType = "icmp"
+)
+
+// TCPMethod represents the type of tcp network to establish.
+type TCPMethod string
+
+const (
+	// TCPMethodPreferSACK represents a preference to sack tcp method if available for network test.
+	TCPMethodPreferSACK TCPMethod = "prefer_sack"
+	// TCPMethodSYN represents a syn tcp method for network test.
+	TCPMethodSYN TCPMethod = "syn"
+	// TCPMethodSACK represents a sack tcp method for network test.
+	TCPMethodSACK TCPMethod = "sack"
+)
+
+// ConfigRequest represents the type configuration for a network test.
+type ConfigRequest interface {
+	GetSubType() SubType
+}
+
+// NetworkConfigRequest represents the generic part of the network test configuration.
+type NetworkConfigRequest struct {
+	SourceService      *string `json:"source_service,omitempty"`
+	DestinationService *string `json:"destination_service,omitempty"`
+	ProbeCount         *int    `json:"probe_count,omitempty"`
+	TracerouteCount    *int    `json:"traceroute_count,omitempty"`
+	MaxTTL             *int    `json:"max_ttl,omitempty"`
+	Timeout            *int    `json:"timeout,omitempty"` // in seconds
+}
+
+// UDPConfigRequest represents a udp network test configuration.
+type UDPConfigRequest struct {
+	Host string `json:"host"`
+	Port *int   `json:"port,omitempty"`
+	NetworkConfigRequest
+}
+
+// GetSubType returns the udp subtype.
+func (u UDPConfigRequest) GetSubType() SubType {
+	return SubTypeUDP
+}
+
+// TCPConfigRequest represents a tcp network test configuration.
+type TCPConfigRequest struct {
+	Host      string    `json:"host"`
+	Port      *int      `json:"port,omitempty"`
+	TCPMethod TCPMethod `json:"tcp_method"` // "prefer_sack" | "syn" | "sack"
+	NetworkConfigRequest
+}
+
+// GetSubType returns the tcp subtype.
+func (t TCPConfigRequest) GetSubType() SubType {
+	return SubTypeTCP
+}
+
+// ICMPConfigRequest represents a icmp network test configuration.
+type ICMPConfigRequest struct {
+	Host string `json:"host"`
+	NetworkConfigRequest
+}
+
+// GetSubType returns the icmp subtype.
+func (i ICMPConfigRequest) GetSubType() SubType {
+	return SubTypeICMP
+}
+
+// SyntheticsTestConfig represents the whole config of a network test.
+type SyntheticsTestConfig struct {
+	Version int    `json:"version"`
+	Type    string `json:"type"`
+	Subtype string `json:"subtype"`
+
+	Config struct {
+		Assertions []interface{} `json:"assertions"`
+		Request    ConfigRequest `json:"request"`
+	} `json:"config"`
+
+	Interval int    `json:"tick_every"`
+	OrgID    int    `json:"orgID"`
+	MainDC   string `json:"mainDC"`
+	PublicID string `json:"publicID"`
+}
+
+// UnmarshalJSON is a Custom unmarshal for SyntheticsTestConfig
+func (c *SyntheticsTestConfig) UnmarshalJSON(data []byte) error {
```

Comment:
Can we unit test this?

In general, when possible, if we have new/modified logic, it should be unit tested.